refactor: deprecate checkhealth functions

dundargoc opened the pull request:
The following functions are deprecated and will be removed in
Nvim v0.11:

- health#report_start()
- health#report_info()
- health#report_ok()
- health#report_warn()
- health#report_error()
- vim.health.report_start()
- vim.health.report_info()
- vim.health.report_ok()
- vim.health.report_warn()
- vim.health.report_error()

Users should instead use the lua versions of these functions (prefixed with `v:lua.` if called from Vimscript):

- vim.health.start()
- vim.health.info()
- vim.health.ok()
- vim.health.warn()
- vim.health.error()

echasnovski commented on runtime/lua/vim/health.lua:
```diff
@@ -1,5 +1,58 @@
 local M = {}
 
+-- Indents lines *except* line 1 of a string if it contains newlines.
+function M._indent_after_line1(s, columns)
+  local lines = vim.split(s, '\n')
+  for i, _ in ipairs(lines) do
+    if i ~= 1 then
+      local indent = ''
+      for _ = 1, columns do
+        indent = ' ' .. indent
+      end
+      lines[i] = indent .. lines[i]
+    end
+  end
+  return table.concat(lines, '\n')
+end
```
This is a more robust approach:
 
```lua
function M._indent_after_line1(s, columns)
  local lines = vim.split(s, '\n')
  local indent = string.rep(' ', columns)
  for i = 2, #lines do
    lines[i] = lines[i]:gsub('^%s*', indent)
  end
  return table.concat(lines, '\n')
end
```

If you want to not indent blank lines, use `^%s*%S` instead of `^%s*`. If you want to not indent only empty lines, use `^%s+`.

dundargoc replied:
Hmm, the string.rep is nice. Thank you!

dundargoc committed:
refactor: deprecate checkhealth functions

The following functions are deprecated and will be removed in
Nvim v0.11:

- health#report_start()
- health#report_info()
- health#report_ok()
- health#report_warn()
- health#report_error()
- vim.health.report_start()
- vim.health.report_info()
- vim.health.report_ok()
- vim.health.report_warn()
- vim.health.report_error()

Users should instead use the lua versions of these functions:

- vim.health.start()
- vim.health.info()
- vim.health.ok()
- vim.health.warn()
- vim.health.error()
